notebooks/search-analysis/search_analysis.ipynb

```diff
@@ -0,0 +1,399 @@
+    "\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "relevancy_judgements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#precision at k (k=5)\n",
+    "no_bestbets = relevancy_judgements[relevancy_judgements['best_bet']==False]\n",
+    "precision = no_bestbets[['query_length','session_id','search_term','score']][no_bestbets['score']>1].groupby(['search_term','session_id','query_length']).count()\n",
+    "precision['score'] = precision['score'] / 5\n",
+    "precision = precision.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Avg Precision at 5 \n",
+    "\n",
+    "from scipy.stats import sem, t\n",
+    "from scipy import mean\n",
+    "avg_score = precision[['query_length','score']].groupby('query_length').agg(['mean','size','sem'])\n",
+    "avg_score = avg_score.droplevel(0,1)\n",
+    "avg_score['h'] = avg_score['sem'] * t.ppf((1 + 0.95) / 2, avg_score['size'] - 1)\n",
+    "avg_score['lower'] = avg_score['mean'] - avg_score['h']\n",
+    "avg_score['upper'] = avg_score['mean'] + avg_score['h']\n",
+    "avg_score = avg_score.reset_index()\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "plt.style.use('ggplot')\n",
+    "plt.figure(figsize=(10,5))\n",
+    "fig, ax = plt.subplots(1, 1)\n",
+    "\n",
+    "plt.plot(avg_score['query_length'], avg_score['mean'], color='green')\n",
+    "ax.fill_between(avg_score['query_length'],avg_score['lower'],avg_score['upper'] , color='b', alpha=.1)\n",
+    "plt.xlabel(\"Query length\")\n",
+    "plt.xticks(rotation=45, ha=\"right\")\n",
+    "plt.ylabel(\"Average Precision @5\")\n",
+    "fig.set_size_inches(10,5)\n",
+    "plt.xlim((1,5))\n",
+    "plt.ylim((0,1))\n",
+    "plt.xticks(np.arange(1, 6, step=1))\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Average max scoring doc in top 5 \n",
+    "\n",
+    "vrel_rate = no_bestbets.groupby('search_term').max('score').reset_index()\n",
+    "vrel_rate = vrel_rate[['search_term','query_length','score']].groupby('query_length').agg(['mean','size','sem'])\n",
+    "vrel_rate = vrel_rate.droplevel(0,1)\n",
+    "vrel_rate['h'] = vrel_rate['sem'] * t.ppf((1 + 0.95) / 2, vrel_rate['size'] - 1)\n",
+    "vrel_rate['lower'] = vrel_rate['mean'] - vrel_rate['h']\n",
+    "vrel_rate['upper'] = vrel_rate['mean'] + vrel_rate['h']\n",
+    "vrel_rate = vrel_rate.reset_index()\n",
+    "vrel_rate\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "%matplotlib inline\n",
```

Comment:
Already called the `matplotlib` inline magic above.
```suggestion

```

Reply:
gone!